config/daemonconfig/mirrors.go

```diff
@@ -0,0 +1,164 @@
+/*
+ * Copyright (c) 2022. Nydus Developers. All rights reserved.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package daemonconfig
+
+import (
+	"fmt"
+	"net/http"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/containerd/containerd/log"
+	"github.com/pelletier/go-toml"
+	"github.com/pkg/errors"
+)
+
+// Copied from containerd, for compatibility with containerd's toml configuration file.
+type hostFileConfig struct {
+	Capabilities []string               `toml:"capabilities"`
+	CACert       interface{}            `toml:"ca"`
+	Client       interface{}            `toml:"client"`
+	SkipVerify   *bool                  `toml:"skip_verify"`
+	Header       map[string]interface{} `toml:"header"`
+	OverridePath bool                   `toml:"override_path"`
+
+	// The following configuration items are specific to nydus.
+	AuthThrough         bool   `toml:"auth_through,omitempty"`
+	HealthCheckInterval int    `toml:"health_check_interval,omitempty"`
+	FailureLimit        uint8  `toml:"failure_limit,omitempty"`
+	PingURL             string `toml:"ping_url,omitempty"`
+}
+
+func isDirExisted(path string) bool {
```

Comment:
We should return an error in this function `isDirExisted`, which might be wrapper to Stat()

Reply:
Yes. Finished.